mitodl/open-discussions
# Update ocw-data-parser, allow list of course paths to be passed to backpopulate_ocw_data

#### Pre-Flight checklist


#### What are the relevant tickets?
Related to https://github.com/mitodl/ocw-studio/issues/1070

#### What's this PR do?
- Updates ocw-data-parser to the latest version
- Modifies the `backpopulate_ocw_data` command to accept lists of course urls (either as strings or in a json file)
- Also ran `isort` on the `course_catalog` app which resorted some imports across various files

#### How should this be manually tested?
- Set `OCW_LEARNING_COURSE_BUCKET_NAME=open-learning-course-data-rc` in your .env file (along with AWS auth settings)
- `docker-compose build`, `docker-compose up`
- Run `manage.py backpopulate_ocw_data --course-url-substring 2-007-design-and-manufacturing-i-spring-2009 --force-s3 --overwrite`
- Check that there are a few dxf files for the course now in S3:
```
import boto3
s3 = boto3.resource("s3")
bucket = s3.Bucket(name="open-learning-course-data-rc")
for obj in bucket.objects.filter(Prefix="2-007-design-and-manufacturing-i-spring-2009"):
    if obj.key.endswith(".dxf"):
        print(obj.key)
```

You can also experiment with the new options for the command:
- Run with ```--course-url-substring 2-007-design-and-manufacturing-i-spring-2009,18-06-linear-algebra-spring-2010```
- Create a `courses.json` file containing some course paths like:
   ```
    [
       "8-591j-systems-biology-fall-2004",
       "15-057-systems-optimization-spring-2003",
       "6-776-high-speed-communication-circuits-spring-2005"
     ]
   ``` 
   and then run with ```--course-url-json courses.json```
